PythonTool/GraveyardRemoval.py

```diff
@@ -33,139 +33,145 @@ def parse_arguments():
-
     tag_name = mb.tag_get_handle(str(types.NAME_TAG_NAME))
 
-    # Gather all entities with a Category tag value of "Group".
+    # Gather all entities with a category tag value of "Group".
     group_categories = get_sets_by_category(mb, "Group")
 
-    # Retrieve all EntitySets with a Name tag.
+    # Retrieve all EntitySets with a name tag.
     group_names = mb.tag_get_data(tag_name, group_categories, flat=True)
 
-    # Find the EntitySet whose Name tag value contains "graveyard".
-    substring = "graveyard"
+    # Find the EntitySet whose name tag value contains "graveyard".
     graveyard_sets = [group_set for group_set, name in zip(group_categories, group_names)
-                      if substring in str(name.lower())]
+                      if "graveyard" in str(name.lower())]
 
-    # Warn the user if there was more than one EntitySet with the "graveyard" Name tag value.
+    # Warn the user if there was more than one EntitySet with the "graveyard" name tag value.
     if len(graveyard_sets) > 1:
         print("WARNING: More than one graveyard set was found.")
 
     # Raise an exception if there was no graveyard EntitySet found.
     if len(graveyard_sets) < 1:
         raise LookupError("WARNING: The geometry file did not contain a graveyard.")
 
-    # If the user would like, print the graveyard entity handle.
-    if print_handle:
-        print("The entity handle(s) of the graveyard EntitySet(s): ")
-        print(graveyard_sets)
-
     # Remove the graveyard EntitySet from the data.
     groups_to_write = [group_set for group_set in group_categories
                        if group_set not in graveyard_sets]
 
+    return groups_to_write, graveyard_sets
+
+
+def format_file_name(input_file, output_file = None):
     """
-    Check if the user specified an output file name.
-    If so, write the file with that name. If not, append onto
-    the original input file name to indicate the graveyard has been removed.
+    Determine the name of the file to write to disk.
+
+    If the user has specified an output file name and extension, use this to
+    write the file. If they haven't, append "_no_grave" onto the name of the
+    input file and add the .h5m extension.
+
+    Input:
+    ______
+       input_file: str
+           User supplied data file location.
+       output_file: str
+           Optional user supplied output file name and extension.
+
+    Returns:
+    ________
+       file_name: str
+           The name of the file to write to disk.
     """
 
     if output_file is not None:
-        mb.write_file(output_file, output_sets=groups_to_write)
+        file_name = output_file
     else:
         input_list = input_file.split("/")
         file_name = '.'.join(input_list[-1].split(".")[:-1])
-        output_file = file_name + "_no_grave.h5m"
-        mb.write_file(output_file, output_sets=groups_to_write)
+        file_name = file_name + "_no_grave.h5m"
 
-    return output_file
+    return file_name
```

Comment:
I don't think the `if output_file is not None:` is required, 
I would just use `if output_file is None:` to form `output_file` and return it